spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/domain/EntityScanner.java

```diff
@@ -42,11 +44,24 @@
 
 	private final ApplicationContext context;
 
+	private final boolean excludeAbstractClasses;
+
 	/**
 	 * Create a new {@link EntityScanner} instance.
 	 * @param context the source application context
 	 */
 	public EntityScanner(ApplicationContext context) {
+		this(context, true);
+	}
+
+	/**
+	 * Create a new {@link EntityScanner} instance.
+	 * @param context the source application context
+	 * @param excludeAbstractClasses true if only concrete classes are scanned, false if
+	 * abstract classes and interfaces are scanned too
+	 */
+	public EntityScanner(ApplicationContext context, boolean excludeAbstractClasses) {
```

Comment:
Considering the use cases of scanning abstract classes is quite specific, I would not surface such an option.